factcast-store-validation/src/test/java/org/factcast/store/pgsql/internal/listen/PgConnectionSupplierTest.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright © 2017-2020 factcast.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.factcast.store.pgsql.internal.listen;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.failBecauseExceptionWasNotThrown;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import java.sql.SQLException;
+import java.util.Properties;
+
+import javax.sql.DataSource;
+
+import org.apache.tomcat.jdbc.pool.PoolConfiguration;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+
+public class PgConnectionSupplierTest {
+
+    PgConnectionSupplier uut;
+
+    @Mock
+    private org.apache.tomcat.jdbc.pool.DataSource ds;
+
+    @BeforeEach
+    void setUp() {
+        initMocks(this);
+        uut = new PgConnectionSupplier(ds);
+    }
+
+    private void setupConnectionProperties(String properties) {
+        PoolConfiguration poolConf = mock(PoolConfiguration.class);
+        when(poolConf.getConnectionProperties()).thenReturn(properties);
+        when(ds.getPoolProperties()).thenReturn(poolConf);
+    }
+
+    @Test
+    void test_wrongDataSourceImplementation() {
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
+            DataSource ds = mock(DataSource.class);
+            new PgConnectionSupplier(ds);
+            failBecauseExceptionWasNotThrown(IllegalStateException.class);
+        });
+    }
+
+    @Test
+    void test_buildCredentials() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The JUnit 4 test method name 'test_buildCredentials' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=4877539)